src/main/java/sx/blah/discord/handle/impl/obj/EmojiImpl.java

```diff
@@ -103,6 +109,32 @@ public String getImageUrl() {
 	}
 
 	@Override
+	public IEmoji changeRoles(IRole... roles) {
+		PermissionUtils.hasPermissions(getGuild(), getClient().getOurUser(), EnumSet.of(Permissions.MANAGE_EMOJIS));
+
+		EmojiObject response = ((DiscordClientImpl) getClient()).REQUESTS.PATCH.makeRequest(String.format(DiscordEndpoints.EMOJIS, getGuild().getStringID()) + getStringID(), new EmojiChangeRequest(getName(), roles), EmojiObject.class);
+		IEmoji newEmoji = DiscordUtils.getEmojiFromJSON(getGuild(), response);
+
+		//TODO save newEmoji in cache?
```

Comment:
This method will not function correctly or as intended. When you change the roles it is expected the internal fields will be changed, but that will not happen in this case. If you trace back to getEmojiFromJSON, unlike the other getXFromJSON methods, *this* object is not mutated to reflect changed state. This must be corrected.